dynamic/clientset/clientset.go

```diff
@@ -175,3 +175,32 @@ func (rc *ResourceClient) RemoveFinalizer(orig *unstructured.Unstructured, name
 		return true
 	})
 }
+
+// AtomicStatusUpdate is similar to AtomicUpdate, except that it updates status.
+func (rc *ResourceClient) AtomicStatusUpdate(resourceMap *dynamicdiscovery.ResourceMap, parentResource *dynamicdiscovery.APIResource, orig *unstructured.Unstructured, update func(obj *unstructured.Unstructured) bool) (result *unstructured.Unstructured, err error) {
+	name := orig.GetName()
+
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		current, err := rc.Get(name, metav1.GetOptions{})
```

Comment:
Add a TODO above this line that ideally we would call `GetStatus` (if it HasSubresource) to respect subresource RBAC rules, but the dynamic client doesn't offer that yet.